lib/score_logic.rb

```diff
@@ -0,0 +1,80 @@
+module Scrabble ##require your gems and classes here # require_relative 'scrabble/whatevs'
+
+  SCOREBOARD = {
+    "A" => 1, "E" => 1, "I" => 1, "O" => 1, "U" => 1,
+    "L" => 1, "N" => 1, "R" => 1, "S" => 1, "T" => 1,
+    "D" => 2, "G" => 2,
+    "B" => 3, "C" => 3, "M" => 3, "P" => 3,
+    "F" => 4, "H" => 4, "V" => 4, "W" => 4, "Y" => 4,
+    "K" => 5,
+    "J" => 8, "X" => 8,
+    "Q" => 10, "Z" => 10,
+  }
+
+  class Scrabble
+
+    def self.score(word)
+
+      return "Must pick a letter!" unless word[/[a-zA-Z]+/] == word && word.length > 0
+      #If the input is a character that isn't a letter, or if it's an empty string, it brings up an error
+
+      get_score_array(word)
+
+      sum = 0
+      # for each letter
+      @score_array.each do |score|
+        sum += score.to_i #adds all the letter scores together
+      end
+      #returns the sum/score of all letters
+      return sum
+    end
+
+    def self.get_score_array(word)
+      word = word.chars
+      # empty array to iterate through each letter of the split word
+      @score_array = []
+      #SCOREBOARD[letter.upcase] gives me the score value from my SCOREBOARD hash
+      word.each do |letter|
+        @score_array.push(SCOREBOARD[letter.upcase])
+      end
+      return @score_array
+    end
+
+    def self.highest_score_from(array_of_words)
+
+      #if the array is empty, it raises an error
+      if array_of_words.empty?
+        puts "Your array of words is empty. Add some words in there!"
+      end
+
+      # if there is a word that contains things other than letters, it brings up an error.
+      array_of_words.each do |word|
+        puts "Use only letters! Try again" unless word[/[a-zA-Z]+/]  == word && word.length > 0
+      end
+
+      #picks highest scoring words from array_of_words
+      highest_score = array_of_words.max_by { |word| score(word) }
+      #picks highest scoring words from array_of_words
+
+      tied_words = array_of_words.select { |word| score(word) == score(highest_score) }
```

Comment:
This could potentially execute the `score(highest_score)` method multiple times so it would make more sense to take it out into a local variable.